collect_app/src/main/java/org/odk/collect/android/tasks/DiskSyncTask.java

```diff
@@ -136,8 +136,11 @@ protected String doInBackground(Void... params) {
                                 uriToUpdate.add(new UriFile(updateUri, sqlFile));
                             }
                         } else {
+                            // remove it from the database also
+                            mFormsDao.deleteFormsForFormId(mCursor.getString(
```

Comment:
I really like the idea of collecting the ids and then deleting these at once. Reaction time of the app will be better, if we can reduce the number of transactions to the database.